lib/service/service.go

```diff
@@ -1229,12 +1230,24 @@ func NewTeleport(cfg *servicecfg.Config) (*TeleportProcess, error) {
 		return nil, trace.Wrap(err)
 	}
 
+	// Check if the deprecated teleport-upgrader script is being used.
```

Comment:
Any chance we could have all auto-upgrader logic into its own function? NewTeleport is already big and complex enough.

Reply:
I moved a few sections out, but I hesitate to do heavier refactoring for this feature. Removing the unit upgrader and single-window scheduling system in the future will clean this up.